adminSiteClient/FilesIndexPage.tsx

```diff
@@ -0,0 +1,373 @@
+                            fontSize: "12px",
+                            color: "#666",
+                            marginTop: 2,
+                        }}
+                    >
+                        {file.path}
+                    </div>
+                )}
+            </div>
+            <Tooltip title="Copy file URL">
+                <button
+                    className="file-viewer__copy-button"
+                    onClick={() => copyToClipboard(fileUrl)}
+                >
+                    <FontAwesomeIcon icon={faCopy} />
+                </button>
+            </Tooltip>
+        </div>
+    )
+}
+
+function FolderButton({
+    name,
+    onClick,
+}: {
+    name: string
+    onClick: () => void
+}) {
+    return (
+        <button className="file-viewer__folder" onClick={onClick}>
+            {name}
+        </button>
+    )
+}
+
+function FileMapViewer({
+    fileMap,
+    currentPath,
+}: {
+    fileMap: FileMap
+    currentPath: string
+}) {
+    const history = useHistory()
+
+    const handleFolderClick = (folderName: string) => {
+        history.push(`?path=${urlJoin(currentPath, folderName)}`)
+    }
+
+    const pathSegments = currentPath.split("/").filter(Boolean)
+    const currentNode = R.pathOr(fileMap, pathSegments as any, fileMap)
+
+    return (
+        <div>
+            {Object.entries(currentNode).map(([key, value]) => {
+                if (checkIsFile(value)) {
+                    return <FileButton key={key} file={value} />
+                } else {
+                    return (
+                        <FolderButton
+                            key={key}
+                            name={key}
+                            onClick={() => handleFolderClick(key)}
+                        />
+                    )
+                }
+            })}
+        </div>
+    )
+}
+
+function SearchResultsViewer({
+    fileMap,
+    searchValue,
+}: {
+    fileMap: FileMap
+    searchValue: string
+}) {
+    const allFiles = useMemo(() => {
+        const flattenFiles = (
+            map: FileMap,
+            currentPath: string = ""
+        ): DbPlainFile[] => {
+            const files: DbPlainFile[] = []
+
+            Object.entries(map).forEach(([key, value]) => {
+                const fullPath = currentPath ? `${currentPath}/${key}` : key
+
+                if (checkIsFile(value)) {
+                    files.push(value)
+                } else {
+                    files.push(...flattenFiles(value, fullPath))
+                }
+            })
+
+            return files
+        }
+
+        return flattenFiles(fileMap)
+    }, [fileMap])
+
+    const filteredFiles = useMemo(() => {
+        return allFiles.filter(
+            (file) =>
+                file.filename
+                    .toLowerCase()
+                    .includes(searchValue.toLowerCase()) ||
+                file.path.toLowerCase().includes(searchValue.toLowerCase())
+        )
+    }, [allFiles, searchValue])
```

Comment:
if you haven't already, I would consider reusing the search utils from `adminShared/search`, which would help with UX consistency across admin UIs (e.g. highlighting)